src/CompanyService.Business/Commands/Department/FindDepartmentsCommand.cs

```diff
@@ -71,30 +71,41 @@ private List<UserData> GetUsers(List<Guid> usersIds, List<string> errors)
 
         public DepartmentsResponse Execute()
         {
-            List<string> errors = new();
+            DepartmentsResponse response = new ();
 
             var dbDepartments = _repository.FindDepartments();
 
-            List<DepartmentInfo> departments = new();
-
             foreach (var department in dbDepartments)
             {
-                UserInfo director = null;
-                List<UserData> usersData = GetUsers(department.Users.Select(x => x.UserId).ToList(), errors);
+                List<Guid> usersIds = department.Users.Select(x => x.UserId).ToList();
+
+                if (department.DirectorUserId != null)
+                {
+                    usersIds.Add((Guid)department.DirectorUserId);
```

Comment:
department.DirectUserId.Value;

Reply:
and before add directorID plz check if list already contains this ID

Reply:
done